tests/compiler/test_plugin_types.py

```diff
@@ -43,6 +43,22 @@ def test_conversion():
 
     assert inmanta_type.TypedDict(inmanta_type.String()) == to_dsl_type(collections.abc.Mapping[str, str])
 
+    # Union types
```

Comment:
Could you add a nested union? e.g. `Union[int, Union[str, float]]`, and perhaps also one with a `None` in the deepest union (`Union[int, Union[str, float None]]`).